xarray/core/common.py

```diff
@@ -1434,7 +1434,7 @@ def _full_like_variable(other, fill_value, dtype: DTypeLike = None):
             other.shape, fill_value, dtype=dtype, chunks=other.data.chunks
         )
     else:
-        data = np.full_like(other, fill_value, dtype=dtype)
+        data = np.full_like(other.data, fill_value, dtype=dtype)
```

Comment:
This LGTM.

The tests are a little hard to review. Is there anything specific you want us to look at?

Reply:
well, the idea is that the tests mostly failed due to the indexes, so I split each test into variants so units in data, dimension coordinates and non-dimension coordinates are tested separately (and we can xfail the dimension coordinates tests).

Other than that, there's the new `weighted` and `pad` tests and I just replaced some `assert np.allclose(...)` statements with a new `assert_duckarray_allclose(...)` function. I think a review could be restricted to just reviewing those new tests / functions.

Reply:
Just two minor comments. The added tests look great!